node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -127,17 +127,18 @@ impl<C: Context> ActiveValidator<C> {
     pub(crate) fn on_new_vote<R: Rng + CryptoRng + ?Sized>(
         &mut self,
         vhash: &C::Hash,
-        timestamp: Timestamp,
+        mut timestamp: Timestamp,
         state: &State<C>,
         instance_id: C::InstanceId,
         rng: &mut R,
     ) -> Vec<Effect<C>> {
         if let Some(evidence) = state.opt_evidence(self.vidx) {
             return vec![Effect::WeEquivocated(evidence.clone())];
         }
-        if self.earliest_vote_time(state) > timestamp {
-            warn!(%timestamp, "skipping outdated confirmation");
```

Comment:
So, with this change, we will never decide that the proposal was too old to confirm?

Reply:
`should_send_confirmation` will return false if the timestamp is not in the same round anymore.

Reply:
I think it was useful, at least for debugging purposes, to see this log messages. It could have been the explanation for the stalled finality. Can you add a corresponding entry? Somewhere…